include/crocoddyl/core/activations/smooth-1norm.hpp

```diff
@@ -51,10 +51,15 @@ class ActivationModelSmooth1NormTpl : public ActivationModelAbstractTpl<_Scalar>
    * @param[in] eps  Smoothing factor (default: 1.)
    */
   explicit ActivationModelSmooth1NormTpl(const std::size_t& nr, const Scalar& eps = Scalar(1.)) : Base(nr), eps_(eps) {
-    if (eps <= Scalar(0.)) {
+    if (eps < Scalar(0.)) {
       throw_pretty("Invalid argument: "
                    << "eps should be a positive value");
     }
+    if (eps == Scalar(0.)) {
+      std::cerr
```

Comment:
Crap. Sorry Carlos, I'm not thinking clearly. I was thinking this was an assert on eps <=0. But this was a throw, and a throw should be valid for this case.

Could you please restore the previous version, and use this new warning in the throw that you just added? Warning: eps=0 leads to derivatives discontinuoties in the origin, it becomes the absolute function

Reply:
hahaha I think you are tired.

You were right before, and not in this statement. Let me explain you :)

Given the following code
```python
import crocoddyl
a = crocoddyl.ActivationModelSmooth1Norm(3, 0.)
```
With the previous implementation, this code fails:
```python
----> 1 a = crocoddyl.ActivationModelSmooth1Norm(3, 0.)

RuntimeError: In /home/cmastalli/devel/crocoddyl/include/crocoddyl/core/activations/smooth-1norm.hpp
crocoddyl::ActivationModelSmooth1NormTpl<Scalar>::ActivationModelSmooth1NormTpl(const size_t&, const Scalar&) [with _Scalar = double; std::size_t = long unsigned int; crocoddyl::ActivationModelSmooth1NormTpl<Scalar>::Scalar = double] 56
Invalid argument: eps should be a positive value
```

Instead with the current code (i.e. using std::cerr), it doesn't fail (only warns you):
```python
Warning: eps=0 leads to derivatives discontinuities in the origin, it becomes the absolute function
```

So, the current code has the desired behavior that you requested.

Reply:
We don't have something better than `std:;cerr` to throw warnings?

Reply:
No, we don't. We are using in a few places `std::cerr`:
```bash
include/crocoddyl/multibody/friction-cone.hxx:    std::cerr << "Warning: nf has to be an even number, set to 4" << std::endl;
include/crocoddyl/multibody/friction-cone.hxx:    std::cerr << "Warning: normal is not an unitary vector, then we normalized it" << std::endl;
include/crocoddyl/multibody/friction-cone.hxx:    std::cerr << "Warning: min_nforce has to be a positive value, set to 0" << std::endl;
include/crocoddyl/multibody/friction-cone.hxx:    std::cerr << "Warning: max_nforce has to be a positive value, set to maximum value" << std::endl;
include/crocoddyl/multibody/wrench-cone.hxx:    std::cerr << "Warning: nf has to be an even number, set to 16" << std::endl;
include/crocoddyl/multibody/wrench-cone.hxx:    std::cerr << "Warning: min_nforce has to be a positive value, set to 0" << std::endl;
include/crocoddyl/multibody/wrench-cone.hxx:    std::cerr << "Warning: max_nforce has to be a positive value, set to maximun value" << std::endl;
include/crocoddyl/core/activations/smooth-1norm.hpp:      std::cerr
include/crocoddyl/core/integrator/euler.hxx:    std::cerr << "Warning: dt should be positive, set to 1e-3" << std::endl;
include/crocoddyl/core/integrator/rk4.hxx:    std::cerr << "Warning: dt should be positive, set to 1e-3" << std::endl;
```

If you know a more efficient way, then let me know. We could handle it in another PR.